spec/controllers/books_controller/index_spec.rb

```diff
@@ -0,0 +1,54 @@
+require 'rails_helper'
+
+RSpec.describe 'API V1 Books', type: :request do
+  context "GET '/api/v1/books'" do
+    let(:json) { JSON.parse(response.body).deep_symbolize_keys }
+    let!(:book) do
+      Book.create(
+        author_name: author_name,
+        title: title,
+        description: description,
+        rating: rating,
+        word_count: word_count
+      )
+    end
+    let(:author_name) { 'J.R.R. Tolkien' }
+    let(:title) { 'The Hobbit' }
+    let(:description) { 'the unforgettable story of Bilbo, a peace-loving hobbit, who embarks on a strange and magical adventure. A timeless classic.' }
+    let(:word_count) { 250_000 }
+    let(:rating) { 4 }
+
+    it 'returns http success' do
+      get api_v1_books_path
+      expect(response).to have_http_status(:success)
+      expect(response.status).to eq(200)
+      expect(json[:data].size).to eq(1)
```

Comment:
Here we're asserting some pretty granular level of details in the response.

Which makes the tests more expensive to maintain over time. If we extended the features/tests at some point it's likely these lines would need editing. As it may know return a different book first etc.

This is an art judging this kind of thing. Judging how much detail to test.

How can we make these assertions less granular and more generic? 

Can we test the structure of the response without testing the book details themselves?

Reply:
Great point. It's good practice for me to work on the app as if it will be maintained over time! Here since we are just testing the response we only need to test if something, anything (as long as it's a book) is returned back? So we would be only testing attributes of the response like status and meta and not the content of the response?

Reply:
Yeah just that. Nice one!

Reply:
pick one Annie